awx/main/utils/plugins.py

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2024 Ansible, Inc.
+# All Rights Reserved.
+
+"""
+This module contains the code responsible for extracting the lists of dynamically generated plugins.
+"""
+
+CLOUD_PROVIDERS = []
+
+
+def discover_available_cloud_provider_plugin_names():
+    global CLOUD_PROVIDERS
+    from awx.main.models.inventory import InventorySourceOptions
+
+    plugin_names = list(InventorySourceOptions.injectors.keys())
+
+    CLOUD_PROVIDERS = plugin_names
```

Comment:
Since this is at the end of the diff, I'm coping the suggested change from https://github.com/ansible/awx/pull/15537/files#r1782889259:
```suggestion
from functools import cache


@cache
def discover_available_cloud_provider_plugin_names() -> list[str]:
    """Return a list of cloud plugin names available in runtime.

    The discovery result is cached since it does not change throughout
    the life cycle of the server run.
    
    :returns: List of plugin cloud names.
    :rtype: list[str]
    """
    from awx.main.models.inventory import InventorySourceOptions

    return list(InventorySourceOptions.injectors.keys())
```